src/pds/ingress/client/pds_ingress_client.py

```diff
@@ -214,7 +219,9 @@ def prepare_batch_for_ingress(ingress_path_batch, prefix, batch_index):
 
         # Calculate the MD5 checksum of the file payload
         with open(ingress_path, "rb") as object_file:
-            md5_digest = hashlib.md5(object_file.read()).hexdigest()
```

Comment:
Weird, my comment on this completely disappeared.

Anyway, to re-iterate: if the file at `ingress_path` is large, then `object_file.read()` will bring the _entire thing_ into memory, which could be a problem. Better to stream it in chunks:
```python
md5 = hashlib.md5()
with open(ingress_path, "rb") as object_file:
    while chunk := object_file.read(4096):
        md5.update(chunk)
md5_digest.hexdigest()
```

Reply:
Great suggestion, will add